### PR TITLE
Replace the RefuseComputationParticipant method with FailComputationParticipant.

### PR DESCRIPTION
**Pull request by SanjayVas**
This can be used by a Duchy to indicate a permanent error, transitioning the parent Computation to the FAILED state.

---
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/91)

### CODE REVIEW

**Review by wangyaopw — APPROVED** [review]

**Review by wangyaopw — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/91)* status: 0 of 8 files reviewed, 1 unresolved discussion (waiting on @efoxepstein and @SanjayVas)

___
*[src/main/proto/wfa/measurement/system/v1alpha/computation_log_entry.proto, line 88 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/91#-MbICbkU7h4yIoX7PO7L:-MbICbkU7h4yIoX7PO7M:bmse44p) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/287c9caed29b25570f32c0c11bd663f5911f5174/src/main/proto/wfa/measurement/system/v1alpha/computation_log_entry.proto#L88)):*
> ```protobuf
> 
>     // Human-readable error message. This should not include any sensitive info.
>     string error_message = 3;
> ```

why is error_message removed? are we writing transient error message into the log_message?

**Review by efoxepstein — CHANGES_REQUESTED** [review]
> Reviewed 8 of 8 files at r1.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/91)* status: all files reviewed, 2 unresolved discussions (waiting on @SanjayVas)

___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/gcloud/spanner/measurement.sdl, line 323 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/91#-MbICVpf5mUBJ6-OvEd2:-MbICVpf5mUBJ6-OvEd3:bj1vafo) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/b9e1ed8095e843e74c2e380681fa0112c20a1d89/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/gcloud/spanner/measurement.sdl#L323)):*
> ```
>   DuchyMeasurementLogDetails BYTES(MAX) NOT NULL,
>   DuchyMeasurementLogDetailsJson STRING(MAX) NOT NULL,
> ) PRIMARY KEY (MeasurementConsumerId, MeasurementId, CreateTime, DuchyId),
> ```

Maybe remove the DuchyId from the PK? All transactions have a unique commit timestamp and we almost certainly don't want to allow inserting logs for multiple duchies in the same transaction.

**Review by SanjayVas — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/91)* status: all files reviewed, 2 unresolved discussions (waiting on @efoxepstein and @wangyaopw)

___
*[src/main/kotlin/org/wfanet/measurement/kingdom/deploy/gcloud/spanner/measurement.sdl, line 323 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/91#-MbICVpf5mUBJ6-OvEd2:-MbIFRZ82WY7sRvHgSZF:bxterse) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/b9e1ed8095e843e74c2e380681fa0112c20a1d89/src/main/kotlin/org/wfanet/measurement/kingdom/deploy/gcloud/spanner/measurement.sdl#L323)):*
<details><summary><i>Previously, efoxepstein (Eli Fox-Epstein) wrote…</i></summary><blockquote>

Maybe remove the DuchyId from the PK? All transactions have a unique commit timestamp and we almost certainly don't want to allow inserting logs for multiple duchies in the same transaction.
</blockquote></details>

Ah, good call. This isn't really a per-Duchy child row, but more like an extension.

Done.
___
*[src/main/proto/wfa/measurement/system/v1alpha/computation_log_entry.proto, line 88 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/91#-MbICbkU7h4yIoX7PO7L:-MbIF_BTBwxTtTqJ37Cg:b-89o5al) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/287c9caed29b25570f32c0c11bd663f5911f5174/src/main/proto/wfa/measurement/system/v1alpha/computation_log_entry.proto#L88)):*
<details><summary><i>Previously, wangyaopw (Yao Wang) wrote…</i></summary><blockquote>

why is error_message removed? are we writing transient error message into the log_message?
</blockquote></details>

Yeah. We only really need one message string for each log entry regardless of whether it's an error.

**Review by efoxepstein — APPROVED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/91)* status: all files reviewed, 1 unresolved discussion (waiting on @wangyaopw)

**Review by wangyaopw — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/91)* status: 7 of 8 files reviewed, all discussions resolved (waiting on @efoxepstein)

**Review by efoxepstein — APPROVED** [review]
> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/91)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @corbantek, @oliver-amzn, and @zachcwc)

**Review by zachcwc — APPROVED** [review]
> Reviewed 7 of 8 files at r1, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/91)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @corbantek and @oliver-amzn)

File: src/main/proto/wfa/measurement/internal/kingdom/measurement_log_entry.proto [hunk 1]
```diff
@@ -43,44 +43,51 @@ message MeasurementLogEntry {
 
     // Time that the error occurred.
     google.protobuf.Timestamp error_time = 2;
+  }
 
+  message Details {
     // Human-readable error message. This should not include any sensitive info.
```
Comment (corbantek) [inline src/main/proto/wfa/measurement/internal/kingdom/measurement_log_entry.proto:49]: Maybe a nit, but is every log_message an error message?

Comment (SanjayVas) [reply src/main/proto/wfa/measurement/internal/kingdom/measurement_log_entry.proto:49]: No, that's a good catch. I copied this comment from elsewhere and forgot to adjust.

**Pull request merged**